### ISSUE TITLE
Just a thought when reading the article on mocking

### ISSUE THREAD

**Issue by justinpage** [SophieDeBenedetto/blog#2]
When learning about mocking, I read the article you've written:

https://www.thegreatcodeadventure.com/mocking-http-requests-in-golang/

As well as an article written by another community member on Medium:

https://blog.learngoprogramming.com/how-to-mock-in-your-go-golang-tests-b9eee7d7c266

I found similar patterns between both articles in using a global variable for swapping the mocks. One alternative I wanted to share was something like the following:

Service: https://gist.github.com/justinpage/9ca61a4b60923306128afdcca1b959b4
Test: https://gist.github.com/justinpage/e138136828fb00552a692d305dd961ea

Basically calling a struct assigned variable as a function within a mocked method. I try to avoid package level variables if possible or those not encapsulated inside a function or method. While this might not be the best pattern for all situations, it was something I was playing with when reading your article and the author above.

Thanks again for writing and blogging!
